common/client/transaction_sender.go

```diff
@@ -0,0 +1,278 @@
+			}()
+			return
+		}
+
+		// Primary Nodes
+		healthyNodesNum++
+		primaryNodeWg.Add(1)
+		go func() {
+			defer primaryNodeWg.Done()
+			result := txSender.broadcastTxAsync(ctx, rpc, tx)
+			select {
+			case <-ctx.Done():
+				return
+			case txResults <- result:
+			}
+
+			select {
+			case <-ctx.Done():
+				return
+			case txResultsToReport <- result:
+			}
+		}()
+	})
+	if err != nil {
+		primaryNodeWg.Wait()
+		close(txResultsToReport)
+		close(txResults)
+		return 0, err
+	}
+
+	// This needs to be done in parallel so the reporting knows when it's done (when the channel is closed)
+	txSender.wg.Add(1)
+	go func() {
+		defer txSender.wg.Done()
+		primaryNodeWg.Wait()
+		close(txResultsToReport)
+		close(txResults)
+	}()
+
+	txSender.wg.Add(1)
+	go txSender.reportSendTxAnomalies(tx, txResultsToReport)
+
+	return txSender.collectTxResults(ctx, tx, healthyNodesNum, txResults)
+}
+
+func (txSender *TransactionSender[TX, CHAIN_ID, RPC]) broadcastTxAsync(ctx context.Context, rpc RPC, tx TX) sendTxResult {
+	txErr := rpc.SendTransaction(ctx, tx)
+	txSender.lggr.Debugw("Node sent transaction", "tx", tx, "err", txErr)
+	resultCode := txSender.txErrorClassifier(tx, txErr)
+	if !slices.Contains(sendTxSuccessfulCodes, resultCode) {
+		txSender.lggr.Warnw("RPC returned error", "tx", tx, "err", txErr)
+	}
+	return sendTxResult{Err: txErr, ResultCode: resultCode}
+}
+
+func (txSender *TransactionSender[TX, CHAIN_ID, RPC]) reportSendTxAnomalies(tx TX, txResults <-chan sendTxResult) {
+	defer txSender.wg.Done()
+	resultsByCode := sendTxErrors{}
+	// txResults eventually will be closed
+	for txResult := range txResults {
+		resultsByCode[txResult.ResultCode] = append(resultsByCode[txResult.ResultCode], txResult.Err)
+	}
+
+	_, _, criticalErr := aggregateTxResults(resultsByCode)
+	if criticalErr != nil {
+		txSender.lggr.Criticalw("observed invariant violation on SendTransaction", "tx", tx, "resultsByCode", resultsByCode, "err", criticalErr)
+		PromMultiNodeInvariantViolations.WithLabelValues(txSender.chainFamily, txSender.chainID.String(), criticalErr.Error()).Inc()
+	}
+}
+
+type sendTxErrors map[SendTxReturnCode][]error
+
+func aggregateTxResults(resultsByCode sendTxErrors) (returnCode SendTxReturnCode, txResult error, err error) {
+	severeCode, severeErrors, hasSevereErrors := findFirstIn(resultsByCode, sendTxSevereErrors)
+	successCode, successResults, hasSuccess := findFirstIn(resultsByCode, sendTxSuccessfulCodes)
+	if hasSuccess {
+		// We assume that primary node would never report false positive txResult for a transaction.
+		// Thus, if such case occurs it's probably due to misconfiguration or a bug and requires manual intervention.
+		if hasSevereErrors {
+			const errMsg = "found contradictions in nodes replies on SendTransaction: got success and severe error"
+			// return success, since at least 1 node has accepted our broadcasted Tx, and thus it can now be included onchain
+			return successCode, successResults[0], fmt.Errorf(errMsg)
```

Comment:
Use errors standard package instead, since you don't need formatting
```suggestion
			return successCode, successResults[0], errors.New(errMsg)
```